apps/web/src/api/api.client.ts

```diff
@@ -84,10 +84,12 @@ function buildUrl(url: string, absoluteUrl: boolean) {
 
 function getHeaders() {
   const token = localStorage.getItem('auth_token');
+  const lastEnvironmentId = localStorage.getItem('novu_last_environment_id');
```

Comment:
Can we throw an error or have a guard that will prevent a situation where environmentId header is missing? Ideally also in the backend. This is extremely important that no API request will be made without the environmentId header to the backend

Reply:
If not mistaken the EnvironmentCommand requires the environment ID using class validator. If not provided, a 400 will be raised for all controllers that need it. 

[See here](https://github.com/novuhq/novu/pull/5768/files#diff-1b0e3f7797aad06baa5dea4639298a417be5d4b075bb84017056e6430c627b9fR6)